File: src/Ui/eshop-genai-ui/src/features/basket/baskets/models/BasketDto.ts

export type BasketItemDto = {
  productId: string
  productName: string
  productPrice: number
  productImageUrl: string
  quantity: number
}

export type BasketDto = {
  id: string
  userId: string
  items: BasketItemDto[]
  createdAt: string
  lastModified: string | null
  ExpirationTime: string | null
}

File: src/Services/Basket/src/Baskets/Dtos/BasketDto.cs
namespace Basket.Baskets.Dtos;

public record BasketDto(
    Guid Id,
    string UserId,
    List<BasketItemsDto> Items,
    DateTime CreatedAt,
    DateTime? LastModified,
    DateTime? ExpirationTime);

public record BasketItemsDto(
    Guid ProductId,
    string ProductName,
    decimal ProductPrice,
    string ProductImageUrl,
    int Quantity);